server/pub/queries.js

```diff
@@ -53,21 +55,24 @@ export const createPub = async ({ communityId, collectionId, ...restArgs }, user
 		pubId: newPub.id,
 	});
 
-	const createCollectionPubs = Promise.all(
-		[...(defaultPubCollections || []), collectionId]
-			.filter((x) => x)
-			.map(async (collectionIdToAdd) => {
-				// defaultPubCollections isn't constrained by the database in any way and might contain IDs
-				// of collections that don't exist, so unfortunately we have to do an existence check here.
-				const collection = await Collection.findOne({ where: { id: collectionIdToAdd } });
-				if (collection) {
-					return createCollectionPub({
-						collectionId: collectionIdToAdd,
-						pubId: newPub.id,
-					});
-				}
-				return null;
-			}),
+	const allCollectionIds = [...(defaultPubCollections || []), ...(collectionIds || [])];
+
+	const createCollectionPubs = Bluebird.each(
```

Comment:
We have to do these in serial because the check that makes CollectionPubs primaries relies on other CollectionPubs that have been created, and a race condition can create two primaries :(